lib/hanami/view/application_context.rb

```diff
@@ -20,18 +22,52 @@ def included(context_class)
 
       def define_initialize
         inflector = application.inflector
+        settings = application[:settings] if application.key?(:settings)
         routes = application[:routes_helper] if application.key?(:routes_helper)
+        assets = application[:assets] if application.key?(:assets)
```

Comment:
Unrelated to the PR but: I think we want to change dry-container so that it doesn't raise when doing `#[]`. It's incompatible with `Hash#[]` and as much as I like strictness, I believe it was not a good idea to have behavior like that. We have such `key?` checks in *many places* across the ecosystem which is clearly an indicator that we want a Hash-like behavior instead. It would be great to have dry-container 1.0 released before H2 final, hence I'm bringing this up /cc @timriley

Reply:
@solnic I feel like the current behaviour is actually right, in that it should be a hard error if you're trying to fetch something from the container that doesn't exist.

With a container typically serving the purpose of vending all the known components of an application or subsystem of some kind, I think that the consequence of resolving an unknown component should be much like the consequence of trying to refer to an unknown class constant — that is to say, an exception. If the container was much looser in this behavior, it would more likely lead to unexpected errors around `nil` objects further down the line, which isn't confidence-building.

It's worth noting that the `key?` checks you're bringing up are typically inside framework/library/integration-style code, rather than in "userland."  I'm definitely willing to trade off the need to check for a key first before resolving in those cases if we can retain the benefits of stricter resolution for the end users of the container.

As for this case in particular, what we're dealing with is a container with _unknowable contents_, which is why we need these checks. The fact that they add a bit of noise here I don't actually mind, because it makes it clearer that this might not even be a registered component in the first place (e.g. kind of like the extra "noise" you might have when you're unwrapping a Maybe monad).